### DISCUSSION TITLE
Bug with excluding dependencies in pyproject.toml

### DISCUSSION THREAD

**Discussion by ollz272** [renovatebot/renovate#22256] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.82.0

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

k8s

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We have the following config in our pyproject.toml file
`ddtrace = ">=1.12.7,!=1.13.*`
This being, we want to exclude the 1.13.* series. We'd expect renovate to respect this, however we keep getting merges with trying to change this, ie:

```
-ddtrace = ">=1.12.7,!=1.13.*"
+ddtrace = ">=1.13.1,!=1.13"
```
We'd expect renovate to only try upgrade us on the 1.12.* series, or 1.14.* series if its available.

### What you tried so far.

We've tried 
`ddtrace = ">=1.12.7,!=1.13"`
`ddtrace = ">=1.12.7,!=1.13.*"`
`ddtrace = ">=1.12.7,<1.13.0"`

But renovate continues with all of these to try upgrade us to the 1.13 series.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Although you see that dependency constraint and reason "clearly they don't want 1.13", to Renovate it's different:
- Currently the project is on 1.12
- 1.13 is available
- Renovate should modify the constraint to allow 1.13

Modifying complex constraints is already hard, there's no way Renovate will be able to parse and understand your constraints to automatically do what you expect.

You either need to:
- Close these PRs when they come (to ignore), or
- Add a package rule with `allowedVersions` telling Renovate to skip 1.13
